orderer/common/server/attestationserver_test.go

```diff
@@ -0,0 +1,319 @@
+			},
+		},
+		Stop: &ab.SeekPosition{
+			Type: &ab.SeekPosition_Specified{
+				Specified: &ab.SeekSpecified{Number: 0},
+			},
+		},
+	}
+	envelope := &cb.Envelope{}
+	payload := &cb.Payload{
+		Header: &cb.Header{
+			ChannelHeader:   protoutil.MarshalOrPanic(channelHeader),
+			SignatureHeader: protoutil.MarshalOrPanic(&cb.SignatureHeader{}),
+		},
+		Data: protoutil.MarshalOrPanic(seekInfo),
+	}
+	envelope.Payload = protoutil.MarshalOrPanic(payload)
+
+	fakeChain := &mock.Chain{}
+	fakeChainManager := &mock.ChainManager{}
+	fakeChainManager.GetChainReturns(fakeChain)
+	fakeInspector := &mock.Inspector{}
+	fakeStreamsOpened := &metricsfakes.Counter{}
+	fakeStreamsOpened.WithReturns(fakeStreamsOpened)
+	fakeStreamsClosed := &metricsfakes.Counter{}
+	fakeStreamsClosed.WithReturns(fakeStreamsClosed)
+	fakeRequestsReceived := &metricsfakes.Counter{}
+	fakeRequestsReceived.WithReturns(fakeRequestsReceived)
+	fakeRequestsCompleted := &metricsfakes.Counter{}
+	fakeRequestsCompleted.WithReturns(fakeRequestsCompleted)
+	fakeBlocksSent := &metricsfakes.Counter{}
+	fakeBlocksSent.WithReturns(fakeBlocksSent)
+	deliverMetrics := &deliver.Metrics{
+		StreamsOpened:     fakeStreamsOpened,
+		StreamsClosed:     fakeStreamsClosed,
+		RequestsReceived:  fakeRequestsReceived,
+		RequestsCompleted: fakeRequestsCompleted,
+		BlocksSent:        fakeBlocksSent,
+	}
+	handler := &deliver.Handler{
+		ChainManager:     fakeChainManager,
+		TimeWindow:       time.Second,
+		BindingInspector: fakeInspector,
+		Metrics:          deliverMetrics,
+		ExpirationCheckFunc: func([]byte) time.Time {
+			return time.Time{}
+		},
+	}
+
+	t.Run("Complete block attestation is successfully", func(t *testing.T) {
+		confSys := genesisconfig.Load(genesisconfig.SampleInsecureSoloProfile, configtest.GetDevConfigDir())
+		genesisBlockSys := encoder.New(confSys).GenesisBlock()
+
+		cryptoProvider, err := sw.NewDefaultSecurityLevelWithKeystore(sw.NewDummyKeyStore())
+		require.NoError(t, err)
+
+		tmpdir, err := ioutil.TempDir("", "attestation_test-")
+		require.NoError(t, err)
+		defer os.RemoveAll(tmpdir)
+
+		lf, _ := newLedgerAndFactory(tmpdir, "testchannelid", genesisBlockSys)
+
+		consenter := &mocks.Consenter{}
+		consenter.HandleChainCalls(handleChainCluster)
+		consenters := map[string]consensus.Consenter{confSys.Orderer.OrdererType: consenter}
+
+		manager := multichannel.NewRegistrar(localconfig.TopLevel{}, lf, mockCrypto(), &disabled.Provider{}, cryptoProvider, nil)
+		manager.Initialize(consenters)
+
+		ledger, err := lf.GetOrCreate("mychannel")
+		require.NoError(t, err)
+
+		genesisBlock := encoder.New(confSys).GenesisBlockForChannel("mychannel")
+		ledger.Append(genesisBlock)
+
+		manager.CreateChain("mychannel")
+		chain := manager.GetChain("mychannel")
+		require.NotNil(t, chain)
+
+		mockstream := NewAttestationServerStream()
+
+		(*handler).ChainManager = &deliverSupport{Registrar: manager}
+
+		asr := &attestationserver{dh: handler, Registrar: manager}
```

Comment:
It's better to call `NewAttestationService` which is what exported, and not directly create the inner struct

Reply:
Updated the code to use exported interface.